include/domain-types.h

```diff
@@ -121,6 +121,28 @@ struct Tag
 
 typedef std::vector<Tag> Tags;
 
+struct Payload
+{
+    Payload(const std::string& fileName)
+        : fileName(fileName)
+        , data(nullptr)
+        , dataSize(0)
+    {
+    }
+
+    Payload(const void* data, size_t dataSize, const std::string& mimeType)
+        : data(data)
+        , dataSize(dataSize)
+        , mimeType(mimeType)
+    {
+    }
+
+    std::string fileName;
+    const void* data;
```

Comment:
2. Yes, comment would be much better, if we leave C style pointer.
3. I also thought about unique_ptr initially. However, I'm not sure if it works with array_ptr. I afraid it is not that's why I suggested array_ptr.
4. Only big issue here is ownership. The smaller issue is style, but we can ignore it if that solves the bigger one.
5. We already decided to ignore module boundary problems (like DLL related).

To keep interface universal, we can keep C style pointer, but you shall have comment that it does not own that memory area.